src/interpreter/validation.rs

```diff
@@ -0,0 +1,85 @@
+use crate::error::InterpreterError;
+use crate::interpreter::Interpreter;
+use fuel_asm::PanicReason;
+use fuel_tx::{Input, Output, Transaction};
+use fuel_types::{AssetId, Word};
+use std::collections::HashMap;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct CheckedTransaction {
+    /// the validated transaction
+    pub(crate) tx: Transaction,
+    /// the initial free balances of the transaction
+    pub(crate) balances: HashMap<AssetId, Word>,
```

Comment:
Nit, but maybe an ordered map like a b tree would be better?

Reply:
Shouldn't be a big deal since we sort this while setting up freebalances in the VM memory.